lib/src/lib.rs

```diff
@@ -53,6 +53,13 @@ pub mod fsmonitor;
 pub mod git;
 #[cfg(feature = "git")]
 pub mod git_backend;
+#[cfg(not(feature = "git"))]
+/// Stub module for when the `git` feature is disabled.
+pub mod git {
+    // TODO FIXME: hack
+    /// Reserved remote name for the backing Git repo.
+    pub const REMOTE_NAME_FOR_LOCAL_GIT_REPO: &str = "git";
+}
```

Comment:
Removing this probably needs a little discussion, I guess, this was just a hack. We probably just want to not use it at all in the given set of commands (like `diff`), but I didn't want to get into a bigger change at the moment, `#[cfg]` is otherwise a little awkward e.g. with chain method syntax.

Reply:
Perhaps some helper (`is_git_remote(&str) -> bool`?) can be moved to `git_util`

Reply:
I thought about that. But you'd still need to `#[cfg]` the use sites in all the command files. But then again, I was wondering if we should even special case the use of the `git` name at all if git support isn't compiled into the binary.

Reply:
I reworked it a bit, but I'm still not fully satisfied.  I put something in `cli_utils` and in `lib` to try and minimize `#[cfg]` clauses, so let me know what you think (maybe make a new comment and resolve this one.)

Reply:
I don't think `#[cfg]` is needed at call sites. I think it's conceptually a `Backend` property like `reserved_remote_name() -> Option<&str>` (but isn't general enough to add trait method.)

Reply:
I think this is resolved for now with the current changes and seems to be the least of all evils.